### ISSUE TITLE
node-library target fails due to missing output-dir in versions 2.0.106 and later

### ISSUE THREAD

**Issue by trieloff** [thheller/shadow-cljs#150]
I have following build config:
```clojure
{ :source-paths ["src"]
  :dependencies [[funcool/httpurr "1.0.0"]
                 [trieloff/clojurescript-openwhisk "0.1.0"]]
  :builds {
      :app {:target :node-script
          :output-dir "target"
          :output-to "target/onto-server.js"
          :main onto.server/main }
      :deprovision {:target :node-library
          :output-to "deprovision.js"
          :output-dir "target"
          :compiler-options {:closure-output-charset "US-ASCII"}
          :exports {:main onto.server/deprovision}}
      :echo {:target :node-library
          :output-to "echo.js"
          :output-dir "target"
          :compiler-options {:closure-output-charset "US-ASCII"}
          :exports {:main onto.server/echo}}
      :provision {:target :node-library
          :output-to "provision.js"
          :output-dir "target"
          :compiler-options {:closure-output-charset "US-ASCII"}
          :exports {:main onto.server/provision}}}}
```

which works fine for `"shadow-cljs": "2.0.105"` but fails in all later versions, starting with `"shadow-cljs": "2.0.106"`. I noticed this by having `npm` pin the last minor version, so even setting to `"shadow-cljs": "^2.0.105"` would cause the build to fail.

The error occurs after flushing optimized modules:

```
shadow-cljs - config: /code/gotta-get-you-onto-my-server/shadow-cljs.edn version: 2.0.106
shadow-cljs - starting ...
[:provision] Compiling ...
[CACHED] cljs/core.cljs
[CACHED] promesa/protocols.cljc
[CACHED] org/bluebird.cljs
[CACHED] promesa/impl.cljc
[CACHED] promesa/impl/scheduler.cljc
[CACHED] promesa/core.cljc
[CACHED] httpurr/protocols.cljc
[CACHED] httpurr/client.cljc
[CACHED] cljs/nodejs.cljs
[CACHED] clojure/string.cljs
[CACHED] httpurr/client/node.cljs
[CACHED] onto/server.cljs
-> Compile CLJS: shadow/umd_helper.cljs
<- Compile CLJS: shadow/umd_helper.cljs (72 ms)
-> Closure - Optimizing ...
<- Closure - Optimizing ... (7097 ms)
-> Flush optimized node script: /code/gotta-get-you-onto-my-server/provision.js
-> Flushing optimized modules
no :output-dir
{}
ExceptionInfo: no :output-dir
	clojure.core/ex-info (core.clj:4739)
	clojure.core/ex-info (core.clj:4739)
	shadow.build.data/output-file (data.clj:238)
	shadow.build.data/output-file (data.clj:235)
	shadow.build.output/flush-optimized/fn--20264 (output.clj:243)
	shadow.build.output/flush-optimized (output.clj:233)
	shadow.build.output/flush-optimized (output.clj:211)
	shadow.build.node/flush-optimized/fn--22254 (node.clj:182)
	shadow.build.node/flush-optimized (node.clj:169)
	shadow.build.node/flush-optimized (node.clj:166)
	shadow.build.targets.node-library/flush (node_library.clj:148)
	shadow.build.targets.node-library/flush (node_library.clj:143)
	shadow.build.targets.node-library/process (node_library.clj:160)
	shadow.build.targets.node-library/process (node_library.clj:150)
	clojure.lang.Var.invoke (Var.java:381)
	shadow.build/process-stage (build.clj:110)
	shadow.build/process-stage (build.clj:102)
	shadow.build/flush (build.clj:316)
	shadow.build/flush (build.clj:312)
	shadow.cljs.devtools.api/release* (api.clj:231)
	shadow.cljs.devtools.api/release* (api.clj:216)
	shadow.cljs.devtools.cli/do-build-command (cli.clj:34)
	shadow.cljs.devtools.cli/do-build-command (cli.clj:30)
	shadow.cljs.devtools.cli/do-build-commands/fn--650/fn--651 (cli.clj:68)
	clojure.core/binding-conveyor-fn/fn--5476 (core.clj:2022)
	java.util.concurrent.FutureTask.run (FutureTask.java:266)
	java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1142)
	java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:617)
	java.lang.Thread.run (Thread.java:748)

```

**Comment by thheller** [comment]
Hmm I just tested this locally and it works as expected. I did a few changes at that version range so it appears that you might be stuck on some old AOT classes or so.

Can you try again after deleting `target/shadow-cljs/aot-classes` and restarting the process?

**Comment by thheller** [comment]
Oops, found it. Fixed in `shadow-cljs@2.0.118`.

The `release` task is only using `:output-to` since it only outputs a single file anyways and doesn't need the dir. Since you are only using a filename without a directory it didn't properly find the "parent". Fixed it to fully resolve the filename first so it has a directory.

**Comment by trieloff** [comment]
Works like a charm. Thank you!
